src/vs/code/electron-main/app.ts

```diff
@@ -363,6 +364,12 @@ export class CodeApplication extends Disposable {
 		const machineId = await this.resolveMachineId();
 		this.logService.trace(`Resolved machine identifier: ${machineId}`);
 
+		let trueMachineId;
```

Comment:
I am super confused now by this code which seems to resolve the machine ID twice depending on some value of the machine ID. I suggest to do this all from within ONE method and not two that does the right thing depending on the state. E.g. `resolveMachineId` should return the "true" machine ID in case needed.

Reply:
@bpasero we need to resolve 2 separate machined ids, not just one. We should always return and use the originally calculated machine id as machineId. TrueMachineId is intended only as a disambiguation in this specific case.